src/frontend/src/optimizer/plan_node/logical_join.rs

```diff
@@ -585,6 +585,12 @@ impl LogicalJoin {
             .and(scan_predicate.rewrite_expr(&mut scan_predicate_rewriter));
 
         let new_join_on = new_eq_cond.and(new_other_cond);
+
+        // We discovered that we cannot use a lookup join after pulling up the predicate
+        // from one side and simplifying the condition. Let's use  some other join instead.
+        if new_join_on.always_false() {
```

Comment:
Maybe the right thing to do is actually to check that the `new_predicate.has_eq`